M2/autogen.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+set -e
+
+if ! git rev-parse --is-inside-work-tree > /dev/null 2>&1
```

Comment:
This is not good enough. e.g. I have track my home directory dot-files with git, so if I extract a tarfile in my home directory, it's still in a git tree. I think you specifically want to check that `../.git` exists as a file or directory.

That said, I still don't think this should straight up fail.

Reply:
Ah, ok -- good point re. `git rev-parse --is-inside-work-tree` v. checking for `../.git`.

Let me make sure I'm clear about your original suggestion.  It basically the following?

```
if inside a clone of the M2 git repo:
    update M2-emacs submodule
else:
    print some sort of warning
```
What would the warning entail?  Right now, it's not possible to build M2 using just a tarball unless you make some modifications ahead of time to avoid the use of git (e.g., for the Debian package, the M2-emacs submodule is vendored inside the tarball and we use Debian packages for everything else).

This script (and the Makefile that preceded it) is designed to be used by folks who have cloned the repository and plan on using the autotools to build M2.  So it really only makes sense to be used inside the git repository.

Reply:
If I extract a tarball, and also extract tarballs of a handful of submodules, then run this script, why would the build system fail? At least the CMake build doesn't, but maybe there's a step in the autotools build that I'm forgetting.